src/store/wallet/wallet-selectors.ts

```diff
@@ -43,30 +42,41 @@ export const useIsAuthorisedSelector = () => {
 
 export const useSelectedAccountSelector = () => useSelector(({ wallet }) => getSelectedAccount(wallet), jsonEqualityFn);
 
-export const useAssetsListSelector = (): TokenInterface[] =>
-  useSelector(state => {
-    const selectedAccountState = getAccountState(state.wallet, state.wallet.selectedAccountPublicKeyHash);
-    const isTezosNode = !isDcpNode(state.settings.selectedRpcUrl);
+export const useAssetsListSelector = (): TokenInterface[] => {
+  const selectedAccountState = useSelectedAccountState();
+
+  const isTezosNode = useSelector(state => !isDcpNode(state.settings.selectedRpcUrl));
+
+  const tokensList = isTezosNode ? selectedAccountState.tokensList : selectedAccountState.dcpTokensList;
+
+  const filteredTokensList = useMemo(
+    () => tokensList.filter(token => selectedAccountState.removedTokensList.indexOf(token.slug) === -1),
+    [tokensList, selectedAccountState]
+  );
 
-    const tokensList = isTezosNode ? selectedAccountState.tokensList : selectedAccountState.dcpTokensList;
+  const getTokenMetadata = useGetTokenMetadata();
 
-    return tokensList
-      .filter(token => selectedAccountState.removedTokensList.indexOf(token.slug) === -1)
-      .map(token => {
+  const result = useMemo(
```

Comment:
You don't need this temporary variable `result`. You can write `return useMemo` like in `useTezosToken` hook